java/serving/src/main/java/feast/serving/registry/RegistryRepository.java

```diff
@@ -16,27 +16,74 @@
  */
 package feast.serving.registry;
 
+import com.google.protobuf.Duration;
 import feast.proto.core.FeatureProto;
 import feast.proto.core.FeatureViewProto;
 import feast.proto.core.OnDemandFeatureViewProto;
 import feast.proto.core.RegistryProto;
 import feast.proto.serving.ServingAPIProto;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
 
-/**
- * RegistryRepository allows the ServingService to retrieve feature definitions from a Registry
- * object. This approach is needed for a feature store created using feast 0.10+.
- */
-public interface RegistryRepository {
-  RegistryProto.Registry getRegistry();
+public class RegistryRepository {
+  private Registry registry;
+  private RegistryFile registryFile;
+
+  public RegistryRepository(RegistryFile registryFile, int refreshIntervalSecs) {
+    this.registryFile = registryFile;
+    this.registry = new Registry(this.registryFile.getContent());
+
+    if (refreshIntervalSecs > 0) {
+      setupPeriodicalRefresh(refreshIntervalSecs);
```

Comment:
In the AWS Lambda Feature server, every `feast apply` is responsible for tearing down existong lambdas and recreating them with the new registry file. Should we follow the same pattern in feast-serving? Just seems weird that the behavior for the two servings options is different in this respect.

Reply:
Although, I agree on consistency point, this is a little different case. Java Feature Server will be deployed in Kubernetes (see my proposal for deploying architecture in #2017 ) with Helm chart (at least for now). I also would argue that restarting pod on each registry change might be not be desired in production environment (not sure whether it's good thing for lambda as well, since it also doesn't start instantly). 
But the main problem that currently feast sdk isn't capable of managing complex infrastructure, even aws lambda is currently deployed not completely (still requires manually configuration), and for this case we would need to manage kubernetes object. Probably that will be solved with introduction of pulumi.
But I still think that this current simple solution deserves to exist. It could be short-term solution until `feast apply` is capable of managing complex infra or long-term for people who will manage infra manually.

Reply:
Okay, yeah I think that's fair. For more "productionized" services, the ability to run without restarts is valuable.

Reply:
Is it worth making a TODO comment here with our reasoning? I just want to make sure that we capture our decision.